### ISSUE TITLE
says “please make sure systems device time

### ISSUE THREAD

**Issue by savyyy001** [mattrubin/authenticator#313]

**Comment by savyyy001** [comment]
Says “please
Make
Sure systems
Device time is correct”

**Comment by beaucollins** [comment]
Is this an alert pop up? Where do you see this message?

My hunch without knowing a lot of context is that since the 6 digit codes are time based, if your device's time is off by enough it's not going to give you the correct codes.

**Comment by savyyy001** [comment]
I get this message, how can I fix the time issue?

**Comment by beaucollins** [comment]
> I get this message, how can I fix the time issue?

Authenticator uses the device's date & time which can be changed via the Settings -> General -> Date & Time:

![Image-1](https://user-images.githubusercontent.com/19795/57944795-f4d70e80-788c-11e9-8087-955e412b53ca.jpeg)

**Comment by savyyy001** [comment]
I put it to this and
Still get the error...
If the app
Is on an old phone will
That be an issue?

**Comment by beaucollins** [comment]
I don't see why the age of the phone should make a difference if it is able to run the app. Which iOS version are you currently running?

**Comment by savyyy001** [comment]
I just updated the system, i meant i have the app on an old phone I am not using.

**Comment by mattrubin** [comment]
@savyyy001 Are you seeing the "please make sure systems device time is correct" message on your phone, or on the website you're trying to log into?
What website are you trying to log into?
Have you successfully logged into that website using Authenticator before?

**Comment by mattrubin** [comment]
Closing this issue due to inactivity. Feel free to add a new comment here, if you're still encountering an issue with the app.

**Comment by mattrubin** [comment]
@savyyy001 please stop posting comments in the other thread (https://github.com/mattrubin/Authenticator/issues/306#issuecomment-500069320) and please rely to the questions I asked here:

 - Are you seeing the "please make sure systems device time is correct" message on your phone, or on the website you're trying to log into?
 - What website are you trying to log into?
 - Have you successfully logged into that website using Authenticator before?

**Comment by savyyy001** [comment]
Not commenting on anything, ive been waiting weeks for a solution. I am trying to access an app using the authenticator app. Im hoping for a solution, this is ridiculous, the time is correct. I live in Arizona and my time is based in AZ. I only use it for snapchat. Im using it on my
IPhone. Please advise asap.

**Comment by beaucollins** [comment]
I took a look at Snapchat’s documentation for two-factor auth.

https://support.snapchat.com/en-US/a/enable-login-verification

It looks like if you don’t have a backup code you are out of luck.

I set up a new Snapchat account, turned on two-factor, removed my device from Snapchat’s settings, and then logged out.

As I signed back in I was greeted with the 2nd factor screen.

I typed in random digits and submitted it.

The result was the error message you’ve been spamming all over the place on this project:

<img width="375" alt="PNG image" src="https://user-images.githubusercontent.com/19795/59357466-c60f5500-8cdf-11e9-971a-5a5f9c415e87.png">

The “system time” is a red herring. It appears you don’t have the correct two factor key for the account you are trying to sign into.

If you don’t have the recovery code Snapchat gives you when you enable two-factor auth, nor have SMS Verification enabled, it sounds as though you’ve lost access to your account per Snapchat’s docs:

> **Important** ⚠️You will be unable to receive a Verification Code or Recovery Code by email. You must have access to your account in order to generate a Recovery Code. It's best to create a Recovery Code as soon as you can, just to be safe! If you have lost access to your account and haven’t created a Recovery Code, then you could be out of luck 😰
>
> Note: We do not offer support for lost Recovery Codes 🚨
